src/command/FileCommands.cxx

```diff
@@ -234,9 +236,93 @@ handle_read_comments(Client &client, Request args, Response &r)
 	gcc_unreachable();
 }
 
+AllocatedPath
+find_art(AllocatedPath directory)
+{
+    std::vector<std::string> art_names = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(std::string& name: art_names) {
+        const AllocatedPath art = AllocatedPath::Build(directory, name.c_str());
+        
+        FileInfo fi;
+        if (!GetFileInfo(art, fi, false)) {
+            continue;
+        }
+        
+        if (!fi.IsRegular()) {
+            continue;
+        }
+        return art;
+    }
+    return AllocatedPath::Null();
+}
+
 CommandResult
 handle_album_art(Client &client, Request args, Response &r)
 {
-    r.Error(ACK_ERROR_SYSTEM, "Unimplemented");
-    return CommandResult::ERROR;
+    assert(args.size == 1 || args.size == 2);
+    
+    const char *const uri = args.front();
+    unsigned long long offset = 0;
+    if (args.size == 2) {
+        offset = args.ParseUnsigned(1);
+    }
+    
+#ifdef ENABLE_DATABASE
+    const Storage *storage = client.GetStorage();
+#endif
+    const auto located_uri = LocateUri(uri, &client
+#ifdef ENABLE_DATABASE
+                                       , storage
+#endif
+                                       );
+    
+    const AllocatedPath path = storage->MapFS(located_uri.canonical_uri);
+    const AllocatedPath directory = path.GetDirectoryName();
+    
+    AllocatedPath art_name = find_art(directory);
+    if (art_name.IsNull()) {
+        if (offset > 0) {
+            r.Error(ACK_ERROR_NO_EXIST, "No art found");
+            return CommandResult::ERROR;
+        }
+        r.Write("size: -1\n");
```

Comment:
If you declare `size: -1` to indicate an error, then I consider this badly designed. You can use magic values (in band signalling) for binary protocols (if they are documented well!), but for text protocols where we have good error reporting (`ACK_ERROR_...`), there is no need for such hacks. Giving a wrong offset is an obvious client error, because the client does know the size already from the first response. So this error condition could be translated to `ACK_ERROR_ARG`.